src/auth/googleauth.ts

```diff
@@ -645,19 +607,13 @@ export class GoogleAuth {
     }
 
     // For GCE, return the service account details from the metadata server
-    const result = await this.transporter.request<CredentialResult>({
-      url:
-          'http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/?recursive=true',
-      headers: {'Metadata-Flavor': 'Google'}
-    });
+    const res = await gcpMetadata.instance(
```

Comment:
teeny nit: no need to abbreviate. Is `res` an abbreviation of `result` or `response`? I think you intended the latter, so let's just use the full name.